bin/hdfcoinc/pycbc_multiifo_add_statmap

```diff
@@ -245,14 +295,228 @@ if not injection_style:
+        sep_bg_data[combo] = sep_bg_data[combo].remove(all_hred_idx)
+
+    hred_ids = []
+    for ifo in all_ifos:
+        hred_ids += list(np.nonzero(abs(combined_bg_data.data['%s/time' % ifo][:] - maxtime) < args.hierarchical_removal_window)[0])
+    combined_bg_data = combined_bg_data.remove(hred_ids)
+    logging.info('Removing {} background triggers from combined background'.format(len(hred_ids)))
+    logging.info("Recalculating individual combination IFARs")
+
+    times_tuple = tuple(combined_fg_data.data[ifo + '/time'][:] for ifo in all_ifos)
+    test_times = np.array([pycbc.events.mean_if_greater_than_zero(tc)[0]
+                           for tc in zip(*times_tuple)])
+    for key in all_ifo_combos:
+        bnlouder, fnlouder = coinc.calculate_n_louder(sep_bg_data[key].data['stat'],
+                                    sep_fg_data[key].data['stat'],
+                                    sep_bg_data[key].data['decimation_factor'])
+        # In principle bg and fg time should be recalculated, but we expect
+        # this to be a very small correction and as such ignore
+        sep_bg_data[key].data['ifar'] = conv.sec_to_year(bg_time[key] / (bnlouder + 1))
+        sep_fg_data[key].data['ifar'] = conv.sec_to_year(bg_time[key] / (fnlouder + 1))
+    logging.info("Recalculating combined IFARs")
+    for key in all_ifo_combos:
+        bnlouder, fnlouder = coinc.calculate_n_louder(sep_bg_data[key].data['stat'],
+                                    combined_fg_data.data['stat'],
+                                    sep_bg_data[key].data['decimation_factor'])
+        far[key] = (fnlouder + 1) / bg_time[key]
+        # Set up variable for whether each coincidence is available in each coincidence time
+        is_in_combo_time[key] = np.zeros(n_triggers)
+        end_times = np.array(f['segments/%s/end' % key][:])
+        start_times = np.array(f['segments/%s/start' % key][:])
+        idx_within_segment = pycbc.events.indices_within_times(test_times,
+                                                               start_times,
+                                                               end_times)
+        is_in_combo_time[key][idx_within_segment] = np.ones_like(idx_within_segment)
+        
+    isincombo_mask = np.array([list(is_in_combo_time[ct]) for ct in all_ifo_combos])
+    fg_fars = np.array([list(far[ct]) for ct in all_ifo_combos])
+
+    # Combine the FARs with the mask to obtain the new ifars
+    combined_fg_data.data['ifar'] = conv.sec_to_year(1. /np.sum(isincombo_mask*fg_fars, axis=0))
+
+
+for combo in all_ifo_combos:
+    final_fg_data[combo] = final_fg_data[combo].__add__(sep_bg_data[combo])
+    for key in final_fg_data[combo].data:
+        full_key = 'foreground/%s/%s' % (combo, key)
+        if full_key in f:
+            del f[full_key]
+        f[full_key] = final_fg_data[combo].data[key]
+
+final_combined_fg = final_combined_fg.__add__(combined_fg_data)
```

Comment:
What is intended here? Also why are you using .__add__ instead of '+' ?

Reply:
The final_combined_fg is intended as a joining of the final_fg_data with the hierarchically removed items, however this was supposed to be being added to an empty DictArray, hence being doubled-up

Reply:
the `+` vs `.__add__` is just a brain fart